conan/tools/ros/rosenv.py

```diff
@@ -36,17 +38,41 @@ def generate(self):
         rosbuildenv = Environment()
         for k, v in self.variables.items():
             rosbuildenv.define(k, v)
-        rosbuildenv.vars(self._conanfile, "build").save_script(self._build_script_file)
+        for build_script_file in [self._build_script_sh_file, self._build_script_bat_file]:
+            rosbuildenv.vars(self._conanfile, "build").save_script(build_script_file)
 
+        #TODO: Add powrshell support generating .ps1 files
+        self._generate_sh_files()
+        self._generate_bat_files()
+
+    def _generate_sh_files(self):
         # Generate conanrosenv.sh script wrapper that calls conanbuild.sh and conanrun.sh
-        # TODO: Windows .bat/.ps1 files still not supported for the wrapper
-        conanbuild_path = os.path.join(self._conanfile.generators_folder, "conanbuild.sh")
-        conanrun_path = os.path.join(self._conanfile.generators_folder, "conanrun.sh")
-        rosenv_wrapper_content = [f". \"{conanbuild_path}\"", f". \"{conanrun_path}\""]
-        create_env_script(self._conanfile, "\n".join(rosenv_wrapper_content),
-                          self._wrapper_script_file, None)
-
-        conanrosenv_path = os.path.join(self._conanfile.generators_folder, self._wrapper_script_file)
-        msg = f"Generated ROSEnv Conan file: {self._wrapper_script_file}\n" + \
+        conanbuild_sh_path = os.path.join(self._conanfile.generators_folder, "conanbuild.sh")
+        conanrun_sh_path = os.path.join(self._conanfile.generators_folder, "conanrun.sh")
+        rosenv_sh_wrapper_content = [f". \"{conanbuild_sh_path}\"", f". \"{conanrun_sh_path}\""]
+        create_env_script(self._conanfile, "\n".join(rosenv_sh_wrapper_content),
+                          self._wrapper_script_sh_file, None)
+
+        conanrosenv_path = os.path.join(self._conanfile.generators_folder,
+                                        self._wrapper_script_sh_file)
+        msg = f"Generated ROSEnv Conan file: {self._wrapper_script_sh_file}\n" + \
               f"Use 'source {conanrosenv_path}' to set the ROSEnv Conan before 'colcon build'"
         self._conanfile.output.info(msg, fg=Color.CYAN)
+
+    def _generate_bat_files(self):
+        # Generate conanrosenv.bat script wrapper that calls conanbuild.bat and conanrun.bat
+        conanbuild_bat_path = os.path.join(self._conanfile.generators_folder, "conanbuild.bat")
```

Comment:
Why generating bat files in Linux? They are useless, in general Conan generates the env-files for the platform only

Reply:
yes, I think it could be refined, but the scripts that ros2 generates like (intall/setup.sh and so) are generated like .sh and .bat regardless of the platform. So I think it aligns with that behavior